### PR TITLE
i18n(pt-BR): Adding `imports.md` and updating `astro-components.md`

### PR DESCRIPTION
**Pull request by caioferrarezi**
This adds the `imports.md` portuguese version and also updates `astro-components.md` based on #522.

### CODE REVIEW

File: src/pages/pt-BR/guides/imports.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import json from './data.json';
+```
+
+Astro da o suporte para a importação de arquivos JSON diretamente em sua aplicação. Arquivos importados retornam o objeto JSON completo no `import` padrão.
+
+## CSS
+
+```js
+// Carrega e injeta o arquivo 'style.css' na página
+import './style.css';
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:87]:
```suggestion
// Carrega e injeta o arquivo 'estilos.css' na página
import './estilos.css';
```

File: src/pages/pt-BR/guides/imports.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Astro da o suporte para a importação de arquivos CSS diretamente em sua aplicação. Estilos importados não proveêm nenhum "export", mas importar um arquivo CSS automaticamente adiciona este estilo à página. Isto funciona para todos os arquivos CSS por padrão e pode suportar também os pré-processadores como Sass e Less via plugins.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:90]:
```suggestion
Astro suporta a importação de arquivos CSS diretamente em sua aplicação. Estilos importados não proveem nenhum "export", mas importar um arquivo CSS irá automaticamente adicionar seus estilos à página. Isto funciona para todos os arquivos CSS por padrão e pode suportar também pré-processadores como Sass e Less via plugins.
```

File: src/pages/pt-BR/guides/imports.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+Se você prefere não escrever CSS, Astro também suporta todas as bibliotecas populares de CSS-in-JS (exemplo: styles-components) para estilização.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:92]:
```suggestion
Se você prefere não escrever CSS, Astro também suporta todas as bibliotecas populares de CSS-in-JS (ex: styled-components) para estilização.
```

File: src/pages/pt-BR/guides/imports.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+## CSS Modules
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:94]:
```suggestion
## Módulos CSS
```

File: src/pages/pt-BR/guides/imports.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+```jsx
+// 1. Converte os nomes das classes de `./style.module.css` para valores únicos e escopados.
+// 2. Retorn um objecto que mapeia os nomes das classes originais aos valores únicos e escopados.
+import styles from './style.module.css';
+
+// Esse exemplo usa JSX, mas você pode usar CSS Modules com qualquer framework.
+return <div className={styles.error}>Sua Mensagem de Erro</div>;
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:102]:
```suggestion
// 1. Converte os nomes das classes de `./estilos.module.css` para valores únicos e escopados.
// 2. Retorna um objeto que mapeia os nomes das classes originais aos seus valores únicos e escopados.
import estilos from './estilo.module.css';
// Esse exemplo usa JSX, mas você pode usar Módulos CSS com qualquer framework.
return <div className={estilos.erro}>Sua Mensagem de Erro</div>;
```

File: src/pages/pt-BR/guides/imports.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Astro da o suporte para CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padão de estilo que mapeia os nomes originais das classes à seus identificadores únicos.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:105]:
```suggestion
Astro suporta Módulos CSS usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretanto, Módulos CSS exportam um objeto padrão que mapeia os nomes originais de suas classes em identificadores únicos.
```

File: src/pages/pt-BR/guides/imports.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+CSS Modules ajudam a reforçar o escopo e o isolamento do componente no frontend com os nomes de classes únicos gerados para sua folha de estilos.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:107]:
```suggestion
Módulos CSS ajudam a reforçar o escopo e o isolamento de componentes no frontend com nomes de classes únicos gerados para suas folhas de estilos.
```

File: src/pages/pt-BR/guides/imports.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+## Outros Assets
+
+```jsx
+import imgReference from './image.png'; // img === '/src/image.png'
+import svgReference from './image.svg'; // svg === '/src/image.svg'
+import txtReference from './words.txt'; // txt === '/src/words.txt'
+
+// Esse exemplo usa JSX, mas você pode importar as referência em qualquer framework
+<img src={imgReference} />;
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:117]:
```suggestion
import referenciaImg from './imagem.png'; // img === '/src/imagem.png'
import referenciaSvg from './imagem.svg'; // svg === '/src/imagem.svg'
import referenciaTxt from './palavras.txt'; // txt === '/src/palavras.txt'

// Esse exemplo usa JSX, mas você pode importar as referências em qualquer framework.
<img src={referenciaImg} />;
```

File: src/pages/pt-BR/guides/imports.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Todos os outros assets que não foram explicitamente mencionados acima podem ser importados via `import` do ESM e irá retornar a URL de referência à construção final do asset. Isto pode ser útil para referenciar assets que não são JavaScript pela URL, como por exemplo, criar um elemento de imagem com o atributo `src` apontando para aquela imagem.
+
+Também pode ser útil colocar as imagens no diretório `public/` como explicado na [página de estutura do projeto](/pt-BR/core-concepts/project-structure/).
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:122]:
```suggestion
Todos os outros assets que não foram explicitamente mencionados acima podem ser importados via `import` do ESM e irão retornar a URL de referência à construção final do asset. Isto pode ser útil para referenciar assets que não são JavaScript pela URL, como por exemplo, criar um elemento `img` com o atributo `src` apontando para aquela imagem.

Também pode ser útil colocar as imagens no diretório `public/` como explicado na [página de estrutura de projetos](/pt-BR/core-concepts/project-structure/).
```

File: src/pages/pt-BR/guides/imports.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+## WASM
+
+```js
+// Carrega e inicializa o arquivo WASM requisitado
+const wasm = await WebAssembly.instantiateStreaming(fetch('/exemplo.wasm'));
+```
+
+Astro da o suporte para carregamento de arquivos WASM (Web Assembly) diretamente na sua aplicação usando a API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) do navegador.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:131]:
```suggestion
Astro suporta o carregamento de arquivos WASM (Web Assembly) diretamente na sua aplicação usando a API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) do navegador.
```

File: src/pages/pt-BR/guides/imports.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+
+## Módulos Integrados Node
+
+Nós recomendamos aos usuários Astro evitar o uso das Módulos integrados do Node.js (`fs`, `path` e etc) sempre quando possível. Astro tem por objetivo ser compatível com múltiplos motores JavaScript no futuro. Isto inclui o [Deno](https://deno.land/) e o [Cloudflare Workers](https://workers.cloudflare.com/) que não possuem suporte aos módulos integrados do Node como o `fs`.
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:136]:
```suggestion
## Módulos Integrados do Node

Nós recomendamos aos usuários do Astro que evitem o uso de módulos integrados do Node.js (`fs`, `path` e etc) sempre que possível. Astro tem o objetivo de ser compatível com múltiplos motores JavaScript no futuro. Isto inclui o [Deno](https://deno.land/) e o [Cloudflare Workers](https://workers.cloudflare.com/) que não possuem suporte aos módulos integrados do Node como o `fs`.
```

File: src/pages/pt-BR/guides/imports.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+Nossa missão é prover alternativas Astro para os módulos Node.js mais comuns. Entretando, isto ainda não existe hoje. Então, se você _realmente_ precisa usar estes módulos, nós não queremos barrá-lo. Astro suporta os módulos Node.js usando o novo prefixo `node:` do Node. Se você precisa ser um arquivo, por exemplo, você pode fazê-lo assim:
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:138]:
```suggestion
Nossa missão é prover alternativas Astro para os módulos Node.js mais comuns. Entretanto, isto estas alternativas ainda não existem hoje. Então, se você _realmente_ precisa utilizar estes módulos, nós não queremos o impedir. Astro suporta os módulos Node.js usando o novo prefixo `node:` do Node. Se você precisa ler um arquivo, por exemplo, você pode fazer assim:
```

File: src/pages/pt-BR/guides/imports.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+---
+// Exemplo: importa o módulo "fs/promises" do Node.js
+import fs from 'node:fs/promises';
+
+const url = new URL('../../package.json', import.meta.url);
+const json = await fs.readFile(url, 'utf-8');
+const data = JSON.parse(json);
+---
+
+<span>Versão: {data.version}</span>
```
Comment (Yan-Thomas) [inline src/pages/pt-BR/guides/imports.md:150]:
```suggestion
const dados = JSON.parse(json);
---
<span>Versão: {dados.version}</span>
```

**Review by Yan-Thomas — COMMENTED** [review]
Hi @caioferrarezi, thanks for your work!! Here are my suggestions. 🚀

File: src/pages/pt-BR/guides/imports.md [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+Se você não viu o tipo de asset que está procurando, veja a nossa [Biblioteca de Integrações](https://astro.build/integrations/). Você pode extender Astro para adicionar o suporte para diferente tipos de arquivos, como componentes Svelte e Vue.
+
+Este guia detalha como os diferentes tipos de assets são construídos pelo Astro e como importá-los corretamente.
+
+Lembre-se que você pode colocar qualquer asset estático no [diretório `public/`](/pt-BR/core-concepts/project-structure/#public) do seu projeto e Astro irá copiá-los diretamente para seu código final. Arquivos do diretório `public/` não são construídos ou empacotado por Astro, o que signifca que qualquer tipo de arquivo é suportado. Você pode referenciar um arquivo do diretório `public/` por um caminho de URL diretamente em seu template HTML.
+
+## JavaScript
+
+```js
+import { getUser } from './user.js';
```
[the remark this answers is not in the document]
Comment (caioferrarezi) [reply src/pages/pt-BR/guides/imports.md:34]: I also tought about doing this, I agree. Can you just edit your suggestion? There's a typo `usario.js` -> `usuario.js`

File: src/pages/pt-BR/guides/imports.md [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+Se você não encontrou o tipo de asset que está procurando, veja a nossa [Biblioteca de Integrações](https://astro.build/integrations/). Você pode expandir o Astro para adicionar suporte a diferente tipos de arquivos, como componentes Svelte e Vue.
+
+Este guia detalha como os diferentes tipos de assets são construídos pelo Astro e como importá-los corretamente.
+
+Lembre-se que você pode colocar qualquer asset estático no [diretório `public/`](/pt-BR/core-concepts/project-structure/#public) do seu projeto e Astro irá copiá-los diretamente em sua construção final. Arquivos do diretório `public/` não são construídos ou empacotados por Astro, o que significa que qualquer tipo de arquivo é suportado. Você pode referenciar um arquivo do diretório `public/` por um caminho de URL diretamente em seus templates HTML.
+
+## JavaScript
+
+```js
+import { getUser } from './user.js';
```
Comment (caioferrarezi) [inline src/pages/pt-BR/guides/imports.md:34]:
```suggestion
import { getUsuario } from './usuario.js';
```

File: src/pages/pt-BR/guides/imports.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (Yan-Thomas) [reply src/pages/pt-BR/guides/imports.md]: Fixed it!

File: src/pages/pt-BR/guides/imports.md (hunk 4 above)
Comment (caioferrarezi) [reply src/pages/pt-BR/guides/imports.md:94]: I don't know if we should translate this, I think this it's a common keyword like React, Vue and etc.

File: src/pages/pt-BR/guides/imports.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+```jsx
+// 1. Converte os nomes das classes de `./estilos.module.css` para valores únicos e escopados.
+// 2. Retorna um objeto que mapeia os nomes das classes originais aos seus valores únicos e escopados.
+import estilos from './estilo.module.css';
+// Esse exemplo usa JSX, mas você pode usar Módulos CSS com qualquer framework.
+return <div className={estilos.erro}>Sua Mensagem de Erro</div>;
+```
+
+Astro da o suporte para CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padão de estilo que mapeia os nomes originais das classes à seus identificadores únicos.
```
Comment (caioferrarezi) [inline src/pages/pt-BR/guides/imports.md:104]:
```suggestion
Astro suporta CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padrão que mapeia os nomes originais das suas classes em identificadores únicos.
```

File: src/pages/pt-BR/guides/imports.md (hunk 4 above)
Comment (Yan-Thomas) [reply src/pages/pt-BR/guides/imports.md:94]: My reasoning for translating CSS Modules, is that, AFAIK, CSS Modules are like a spec for different build and processing tools to implement, although not an official one. For example, `Selectors Level 4` are translated to `Seletores Nível 4` on a lot of websites I found. I don't really have strong opinions on this, feel free to decide.

Comment (caioferrarezi) [reply src/pages/pt-BR/guides/imports.md:94]: That's alright, I'll keep the translation! If we chang our mind, we can update this later.

**Comment by caioferrarezi** [comment]
@Yan-Thomas! I think I've finished to commit your suggestions. Thanks for reviewing!

**Review by Yan-Thomas — APPROVED** [review]

**Pull request merged**